Updated Dapr Agents docs to align with recent quickstart changes (#4987)
dapr/docs: daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md

### 1. Clone the repository and examine its content

```bash
git clone https://github.com/dapr/dapr-agents.git
cd dapr-agents/quickstarts/01-dapr-agents-fundamentals
```

### 2. Create a virtual environment and install dependencies

From the `01-dapr-agents-fundamentals` folder, do:

```bash
python3.10 -m venv .venv

# Activate the virtual environment 
# On Windows:
.venv\Scripts\activate
# On macOS/Linux:
source .venv/bin/activate

# Install dependencies from the quickstart
pip install -r requirements.txt
```

This installs `dapr-agents` and any additional libraries needed by the examples.

## Understand the application

This example creates an agent that assists with weather information and uses Dapr to handle LLM interactions, persist conversation history, and provide reliable, durable execution of the agent’s steps.

For this quickstart you’ll primarily work with:

* `06_durable_agent_http.py` – the main durable weather agent application exposed over HTTP
* `function_tools.py` – contains `slow_weather_func`, the tool used by the agent
* `resources/llm-provider.yaml` – Conversation API and LLM configuration

> * `resources/conversation-statestore.yaml` – conversation memory state store

Reviewer: Should this be called conversation or just memory store? And where does the term session come into this?

Author: Sessions are like the different threads in ChatGPT left panel.All sessions are stored in the conversation memory.

* `resources/workflow-statestore.yaml` – workflow and durable execution state store

Open `06_durable_agent_http.py`:

```python
from dapr_agents.llm import DaprChatClient

from dapr_agents import DurableAgent
from dapr_agents.agents.configs import AgentMemoryConfig, AgentStateConfig
from dapr_agents.memory import ConversationDaprStateMemory
from dapr_agents.storage.daprstores.stateservice import StateStoreService
from dapr_agents.workflow.runners import AgentRunner
from function_tools import slow_weather_func

def main() -> None:
    # This agent is of type durable agent where the execution is durable
    weather_agent = DurableAgent(
        name="WeatherAgent",
        role="Weather Assistant",
        instructions=["Help users with weather information"],
        tools=[slow_weather_func],
        # Configure this agent to use Dapr Conversation API.
        llm=DaprChatClient(component_name="llm-provider"),
        # Configure the agent to use Dapr State Store for conversation history.
        memory=AgentMemoryConfig(
            store=ConversationDaprStateMemory(

>                 store_name="conversation-statestore",

Reviewer: All the names here should match those that we agree as the default names. Should this not be "agent-memory" ?

Author: This doc is aligned with the quickstart. 
Default component names we introduce recently and quickstarts are not updated with them.
When the quickstarts are updated with the default component names, we'll update the doc as well.

                session_id="06-durable-agent-http",
            )
        ),
        # This is where the execution state is stored

>         state=AgentStateConfig(

Reviewer: I actually think this class should be renamed to AgentWorkflowConfig() and not AgentStateConfig() so it is explicit

Author: Note taken. But this will be done separately from this docs PR.

            store=StateStoreService(store_name="workflow-statestore"),
        ),
    )

>     # This runner will run the agent and expose it on port 8001

Reviewer: I would love to have a longer more detailed description of what a runner can do here. in particular is this 1:1 to an agent?

>     # This runner will run the agent and expose it on port 8001

Reviewer: ```suggestion
    # The AgentRunner expose the weather agent on HTTP port 8001 using the serve capability. You could also expose the weather agent as standalone or to send message with XXX" <- Complete this
```

Author: Done

    # AgentRunner exposes the weather agent over HTTP on port 8001 using serve.
    # The same runner supports PubSub subscriptions and direct in-process invocation.

In the agent definition:

```python
llm=DaprChatClient(component_name="llm-provider"),
```

This uses [Dapr Conversation API]({{% ref "conversation-overview" %}}) via the `llm-provider` component. The corresponding Dapr component is defined in `resources/llm-provider.yaml`:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: llm-provider
spec:
  type: conversation.openai
  version: v1
  metadata:
  - name: key
    value: "{{OPENAI_API_KEY}}"
  - name: model
    value: gpt-4.1-2025-04-14
```

* The `conversation.openai` component type configures the LLM provider and model.
* `key` holds the API key used to authenticate with the LLM provider.

Replace `{{OPENAI_API_KEY}}` with your actual API key so the Conversation API can perform chat completion.

With this setup, you can swap models or even providers by editing the component YAML without changing the agent code.

### Conversation memory with a Dapr state store

In the agent definition, conversation memory is configured as:

```python
memory=AgentMemoryConfig(
  store=ConversationDaprStateMemory(
      store_name="conversation-statestore",
      session_id="06-durable-agent-http",
  )
),
```

> This tells the agent to store conversation history in a Dapr state store named `conversation-statestore`, under a given `session_id`. The matching Dapr component is `resources/conversation-statestore.yaml`:

Reviewer: See comment above about renaming this to memory-statestore

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:

>   name: conversation-statestore

Reviewer: We should use the same as the default names here IMO. So `agent-memory`

spec:
  type: state.redis
  version: v1
  metadata:
    - name: redisHost
      value: localhost:6379
    - name: redisPassword
      value: ""
```

* The state store uses Redis to persist conversation turns.
* The agent reads and writes messages here so the LLM can maintain context across multiple HTTP calls.

You can browse this state later (for example, with Redis Insight) to see how conversation history is stored.

### Durable execution state with a workflow state store

The agent’s durable execution state is configured as:

```python
state=AgentStateConfig(
  store=StateStoreService(store_name="workflow-statestore"),
),
```

This uses a Dapr state store named `workflow-statestore` to persist workflow and agent execution state. The corresponding component is `resources/workflow-statestore.yaml`:

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:

>   name: workflow-statestore

Reviewer: We should use the same as the default names here IMO. So `agent-workflow` (although I do prefer `agent-workflow-statestore` to be more explicit.

spec:
  type: state.redis
  version: v1
  metadata:
  - name: redisHost
    value: localhost:6379
  - name: redisPassword
    value: ""
  - name: actorStateStore
    value: "true"
```

> * This is again a Redis-backed store holds durable workflow state.

Reviewer: ```suggestion
* This is another Redis state store that holds the durable workflow state.
```

   ```

You’ll see that the workflow continues from its last persisted step instead of starting over. The tool call or LLM calls are not re-executed unless required, and you do not need to send a new prompt. Once the workflow completes, the GET request returns the final result.

In summary, the Dapr Workflow engine preserves the execution state of the agent across restarts, enabling reliable long-running interactions that combine LLM calls, tools, and stateful reasoning.
 
## Inspect workflow executions with Diagrid Dashboard

After starting the durable agent with Dapr, you can use the local [Diagrid Dashboard](https://diagrid.ws/diagrid-dashboard-docs) to visualize and inspect your workflow state, including detailed execution history for each run. The dashboard runs as a container and connects to the same state store used by Dapr workflows (by default, the local Redis instance).

<img src="/images/workflow-overview/workflow-diagrid-dashboard.png" width=800 alt="Diagrid Dashboard showing local workflow executions"/><br/>

Start the Diagrid Dashboard container using Docker:

```bash
docker run -p 8080:8080 ghcr.io/diagridio/diagrid-dashboard:latest
```

> Open the dashboard in a browser at [http://localhost:8080](http://localhost:8080) to explore your local workflow executions.

Reviewer 2: ```suggestion
Open the dashboard in a browser at `http://localhost:8080` to explore your local workflow executions.
```
